oracle/src/poseidon.rs

```diff
@@ -48,7 +48,7 @@ pub struct ArithmeticSponge<F: Field, SC: SpongeConstants> {
     // TODO(mimoo: an array enforcing the width is better no? or at least an assert somewhere)
     pub state: Vec<F>,
     params: ArithmeticSpongeParams<F>,
-    pub constants: std::marker::PhantomData<SC>,
+    constants: std::marker::PhantomData<SC>,
```

Comment:
here again `constants` are only accessed in creation time and there's a `new` for `ArithmeticSponge`